cpp/include/rapidsmpf/buffer/buffer.hpp

```diff
@@ -105,6 +111,28 @@ class Buffer {
      */
     [[nodiscard]] void const* data() const;
 
+    /**
+     * @brief ensure all stream-ordered work to populate the buffer is completed.
+     */
+    void synchronize() const;
```

Comment:
Ok, but how do we prevent abuse then? In my mind, providing an interface like this is recipe to "go do the easy thing".